### PR TITLE
Function module

### PR DESCRIPTION
**Pull request by jdahlblom**
Once previous PR are merged I will go through all files and make them point to this.

### CODE REVIEW

File: Scripts/DCS-BIOS/lib/common/Functions.lua [hunk 1]
```diff
@@ -1,10 +1,41 @@
...
+	str = tostring(Functions.Coerce_nil_to_string(str))
+    return string.rep(' ', len - #str)..str
+end
+
+
+return Functions
+
+
+
+
```
Comment (charliefoxtwo) [inline Scripts/DCS-BIOS/lib/common/Functions.lua:41]: nit: there are a _lot_ of extra newlines at the end here

File: Scripts/DCS-BIOS/lib/common/Functions.lua [hunk 2]
```diff
@@ -1,10 +1,41 @@
+module("Functions", package.seeall)
 
----@func Returns the path of the calling script
+local Functions = {}
+
+
+---Returns the path of the calling script
 ---@return string
-function ScriptPath()
+function Functions.ScriptPath()
```
Comment (charliefoxtwo) [inline Scripts/DCS-BIOS/lib/common/Functions.lua:8]: As these are no longer global functions, their names should start with a lowercase letter correct?

File: Scripts/DCS-BIOS/lib/common/Functions.lua [hunk 3]
```diff
@@ -1,10 +1,41 @@
...
+
+ ---@func Returns value if not nil or an empty string
+ ---@param value string
+ ---@return string
+ function Functions.Coerce_nil_to_string(value)
+	if value == nil then
+		return ""
+	else
+		return value
+	end
```
Comment (charliefoxtwo) [inline Scripts/DCS-BIOS/lib/common/Functions.lua:21]:
```suggestion
	return value and value or ""
```

File: Scripts/DCS-BIOS/lib/common/Functions.lua [hunk 4]
```diff
@@ -1,10 +1,41 @@
...
+end
+
+ --Constructs a string of the specified length with the left padded by whitespace if necessary.
+---@param str string The base text
+---@param len number The length the string should be
+---@return string result A new string of length len, with whitespace padding added to the left as necessary
+function Functions.PadLeft(str, len)
+	str = tostring(Functions.Coerce_nil_to_string(str))
```
Comment (charliefoxtwo) [inline Scripts/DCS-BIOS/lib/common/Functions.lua:29]: `str` is documented here as non-nullable, so there's no need to call `coerce_nil_to_string`. If `str` _can_ be `nil`, then we should update the documentation to `string?`

nitpick: this line seems to be double-indented?

**Review by charliefoxtwo — COMMENTED** [review]
It looks like stylua isn't running on this folder - we should probably fix that?

File: Scripts/DCS-BIOS/lib/common/Functions.lua (hunk 1 above)
Comment (jdahlblom) [reply Scripts/DCS-BIOS/lib/common/Functions.lua]: Yeah I thought I'd add extra but it was apparently only *one* extra line.

**Pull request merged**